src/IECoreScene/MeshAlgoReorder.cpp

```diff
@@ -68,35 +69,42 @@ struct ReorderFn
 {
 		std::string m_name;
 
-		ReorderFn( const std::vector<int> &remapping ) : m_remapping( remapping )
+		ReorderFn( const std::vector<int> &remapping ) : m_remapping( remapping ), m_dataIndices( nullptr )
 		{
 		}
 
-		template<typename T>
-		DataPtr operator()( const TypedData<std::vector<T> > *d )
+		void setDataIndices( const IECore::IntVectorData *dataIndices )
+		{
+			m_dataIndices = dataIndices ? &dataIndices->readable() : nullptr;
+		}
+
+		template<typename T, template<typename> class V>
+		PrimitiveVariableAlgos::IndexedData operator()( const V<std::vector<T> > *d )
 		{
 			const auto &inputs = d->readable();
+			IECoreScene::PrimitiveVariable::IndexedView<T> dataView( inputs, m_dataIndices );
 
-			typename TypedData<std::vector<T> >::Ptr data = d->copy();
-			auto &outputs = data->writable();
+			PrimitiveVariableAlgos::IndexedPrimitiveVariableBuilder<T, V> builder( inputs.size(), m_dataIndices ? m_dataIndices->size() : 0, d );
```

Comment:
I feel like I'm missing something, since I would assume we don't need anything as complex as an IndexedPrimitiveVariable builder here. I thought it would be sufficient to simply reorder the indices instead of the data, which is what we do in [`reverseWinding()`](https://github.com/ImageEngine/cortex/blob/master/src/IECoreScene/MeshAlgoWinding.cpp#L99). Can you put my mind at rest and explain why that's not sufficient in this case?

Reply:
I think I was worried about the case where 2 prim vars are sharing indices, and the builder just seemed easy to use.

Reply:
> I think I was worried about the case where 2 prim vars are sharing indices

I think that would be taken care of by the `d->copy()` in `ReorderFn` wouldn't it?

> the builder just seemed easy to use

My understanding is that the builder is only needed when we're changing the number of elements in `PrimitiveVariable::data`, and need to adjust the values of the indices accordingly. If I understand correctly, this is a much simpler case where the data remains unchanged and we just need to reorder the indices, in which case we should avoid any unnecessary overhead or complexity.

Reply:
Addressed in 718c7a4eb8b17240612b37e47e67e2252df5e754

Reply:
Much simpler - thanks!